rusk-profile/src/lib.rs

```diff
@@ -6,148 +6,45 @@
 }
 
 impl Keys {
-    pub fn get_dir(&self) -> Option<PathBuf> {
-        if let Ok(mut dir) = get_rusk_keys_dir() {
-            info!("Found the rusk keys dir");
-
-            dir.push(&self.crate_name);
-            dir.push(&self.version);
-            Some(dir)
-        } else {
-            info!("Couldn't found the rusk keys dir");
-
-            None
-        }
-    }
-
-    pub fn are_outdated(&self) -> bool {
-        let outdated = self.get_dir().map_or(true, |dir| !dir.exists());
-        info!("keys outdated: {}", outdated);
-        outdated
-    }
-
-    pub fn get(&self, handle: &str) -> Option<(Vec<u8>, Vec<u8>)> {
-        info!("Getting VK and PK for handle: {}", handle);
-        let dir = self.get_dir();
-
-        dir.filter(|dir| dir.exists()).and_then(|dir| {
-            let mut hasher = Sha256::new();
-            hasher.update(handle.as_bytes());
-            let hash = format!("{:x}", hasher.finalize());
-
-            let mut pk_file = dir.clone();
-            pk_file.push(format!("{}.pk", hash));
-            let mut vk_file = dir;
-            vk_file.push(format!("{}.vk", hash));
-
-            let pk = read(pk_file);
-            let vk = read(vk_file);
-
-            match (pk, vk) {
-                (Ok(pk), Ok(vk)) => {
-                    info!("Found VK and PK in the cache");
-                    Some((pk, vk))
-                }
-                (_, _) => {
-                    info!("No VK and PK are present in the cache");
-                    None
-                }
-            }
-        })
-    }
-
-    pub fn get_prover(&self, handle: &str) -> Option<Vec<u8>> {
-        let dir = self.get_dir();
-
-        dir.filter(|dir| dir.exists()).and_then(|mut dir| {
-            let mut hasher = Sha256::new();
-            hasher.update(handle.as_bytes());
-            let hash = format!("{:x}", hasher.finalize());
-
-            dir.push(format!("{}.pk", hash));
-
-            read(dir).ok()
-        })
-    }
-
-    pub fn get_verifier(&self, handle: &str) -> Option<Vec<u8>> {
-        let dir = self.get_dir();
-
-        dir.filter(|dir| dir.exists()).and_then(|mut dir| {
-            let mut hasher = Sha256::new();
-            hasher.update(handle.as_bytes());
-            let hash = format!("{:x}", hasher.finalize());
-
-            dir.push(format!("{}.vk", hash));
-
-            read(dir).ok()
-        })
-    }
-
-    pub fn clear_all(&self) -> Result<(), io::Error> {
-        info!(
-            "Clearing all the keys for any version of {}",
-            &self.crate_name
-        );
+    pub fn get_prover(&self) -> Result<Vec<u8>, io::Error> {
         let mut dir = get_rusk_keys_dir()?;
-
-        dir.push(&self.crate_name);
-        if dir.exists() {
-            fs::remove_dir_all(dir)?;
-            info!("Keys removed from the cache");
-        } else {
-            info!("Noop, the folder didn't exist in the cache");
+        dir.push(format!("{}.pk", hex::encode(self.id)));
```

Comment:
We should use Rust API to push extensions and filenames. Check:
https://doc.rust-lang.org/std/path/struct.PathBuf.html#method.set_extension

This will guarantee interop with different platforms

Reply:
You're right!